clients/filesystem-fuse/src/default_raw_filesystem.rs

```diff
@@ -0,0 +1,398 @@
+        self.file_entry_manager.write().await.insert(
+            Self::ROOT_DIR_PARENT_FILE_ID,
+            Self::ROOT_DIR_FILE_ID,
+            Self::ROOT_DIR_NAME,
+        );
+        self.fs.init().await
+    }
+
+    async fn get_file_path(&self, file_id: u64) -> String {
+        let file_entry = self.get_file_entry(file_id).await;
+        file_entry
+            .map(|x| x.path)
+            .unwrap_or_else(|_| "".to_string())
+    }
+
+    async fn valid_file_handle_id(&self, file_id: u64, fh: u64) -> crate::filesystem::Result<()> {
+        let fh_file_id = self
+            .opened_file_manager
+            .get(fh)
+            .ok_or(Errno::from(libc::EBADF))?
+            .lock()
+            .await
+            .file_stat
+            .file_id;
+
+        (file_id == fh_file_id)
+            .then_some(())
+            .ok_or(Errno::from(libc::EBADF))
+    }
+
+    async fn stat(&self, file_id: u64) -> crate::filesystem::Result<FileStat> {
+        let file_entry = self.get_file_entry(file_id).await?;
+        let mut file_stat = self.fs.stat(&file_entry.path).await?;
+        file_stat.set_file_id(file_entry.parent_file_id, file_entry.file_id);
+        Ok(file_stat)
+    }
+
+    async fn lookup(&self, parent_file_id: u64, name: &str) -> crate::filesystem::Result<FileStat> {
+        let parent_file_entry = self.get_file_entry(parent_file_id).await?;
+        let mut file_stat = self.fs.lookup(&parent_file_entry.path, name).await?;
+        // fill the file id to file stat
+        self.resolve_file_id_to_filestat(&mut file_stat, parent_file_id)
+            .await;
+        Ok(file_stat)
+    }
+
+    async fn read_dir(&self, file_id: u64) -> crate::filesystem::Result<Vec<FileStat>> {
+        let file_entry = self.get_file_entry(file_id).await?;
+        let mut child_filestats = self.fs.read_dir(&file_entry.path).await?;
+        for file in child_filestats.iter_mut() {
+            self.resolve_file_id_to_filestat(file, file.file_id).await;
+        }
+        Ok(child_filestats)
+    }
+
+    async fn open_file(&self, file_id: u64, flags: u32) -> crate::filesystem::Result<FileHandle> {
+        self.open_file_internal(file_id, flags, FileType::RegularFile)
+            .await
+    }
+
+    async fn open_dir(&self, file_id: u64, flags: u32) -> crate::filesystem::Result<FileHandle> {
+        self.open_file_internal(file_id, flags, FileType::Directory)
+            .await
+    }
+
+    async fn create_file(
+        &self,
+        parent_file_id: u64,
+        name: &str,
+        flags: u32,
+    ) -> crate::filesystem::Result<FileHandle> {
+        let parent_file_entry = self.get_file_entry(parent_file_id).await?;
+        let mut opened_file = self
+            .fs
+            .create_file(&parent_file_entry.path, name, OpenFileFlags(flags))
+            .await?;
+
+        opened_file.set_file_id(parent_file_id, self.next_file_id());
+
+        // insert the new file to file entry manager
+        {
```

Comment:
use a separate function to do this?

Reply:
Just two lines of code ,it might not be necessary to extract a method.

Reply:
the function is used to encapsulate the details like acquiring the lock

Reply:
Using a code block here is also a common practice.